### ISSUE TITLE
Cannot (re-)bootstrap internationalized app packages

### ISSUE THREAD

**Issue by jpnurmi** [invertase/melos#101]
`melos bootstrap` tries to run `flutter pub get` in the synthetic package (`.dart_tool/flutter_gen`) of [internationalized](https://flutter.dev/docs/development/accessibility-and-localization/internationalization) Flutter apps, which results to:

```
melos bootstrap
   └> /home/jpnurmi/Temp/melos_gen_l10n

Running "flutter pub get" in workspace packages...

------------------------------------------------------------------------------------------------------------
Running "flutter pub get" in flutter_gen...                     
pubspec.yaml has no lower-bound SDK constraint.
You should edit pubspec.yaml to contain an SDK constraint:
environment:
  sdk: '>=2.10.0 <3.0.0'
See https://dart.dev/go/sdk-constraint
pub get failed (65; See https://dart.dev/go/sdk-constraint)
• synthetic_package
    └> ./packages/gen_l10n_example/.dart_tool/flutter_gen
    └> Failed to run "flutter pub get" in this package.
------------------------------------------------------------------------------------------------------------

Bootstrap failed: "flutter pub get" failed in one of your workspace packages.
```

<details><summary>Detailed steps to reproduce the problem with gen_l10n_example from flutter/website</summary>

```
$ mkdir melos_gen_l10n
$ cd melos_gen_l10n/
$ cat <<EOT >> melos.yaml
name: melos_gen_l10n
packages:
  - packages/**
environment:
  sdk: ">=2.12.0 <3.0.0"
EOT
$ mkdir packages
$ git clone --depth=1 https://github.com/flutter/website.git
Cloning into 'website'...
remote: Enumerating objects: 2296, done.
remote: Counting objects: 100% (2296/2296), done.
remote: Compressing objects: 100% (1922/1922), done.
remote: Total 2296 (delta 199), reused 1838 (delta 165), pack-reused 0
Receiving objects: 100% (2296/2296), 251.46 MiB | 14.88 MiB/s, done.
Resolving deltas: 100% (199/199), done.
Updating files: 100% (2060/2060), done.
$ cp -r website/examples/internationalization/gen_l10n_example packages/
$ cd packages/gen_l10n_example
$ flutter pub get
Running "flutter pub get" in gen_l10n_example...
$ cat .dart_tool/flutter_gen/pubspec.yaml
# Generated by the flutter tool
name: synthetic_package
description: The Flutter application's synthetic package.
$ cd -
/home/jpnurmi/Temp/melos_gen_l10n
$ melos bootstrap
melos bootstrap
   └> /home/jpnurmi/Temp/melos_gen_l10n

Running "flutter pub get" in workspace packages...

--------------------------------------------------------------------------------
Running "flutter pub get" in flutter_gen...                     
pubspec.yaml has no lower-bound SDK constraint.
You should edit pubspec.yaml to contain an SDK constraint:
environment:
  sdk: '>=2.10.0 <3.0.0'
See https://dart.dev/go/sdk-constraint
pub get failed (65; See https://dart.dev/go/sdk-constraint)
• synthetic_package
    └> ./packages/gen_l10n_example/.dart_tool/flutter_gen
    └> Failed to run "flutter pub get" in this package.
--------------------------------------------------------------------------------

Bootstrap failed: "flutter pub get" failed in one of your workspace packages.
```
</details>

<details><summary>flutter doctor -v</summary>

```
$ flutter doctor -v
[✓] Flutter (Channel master, 2.3.0-1.0.pre.168, on Linux, locale en_DK.UTF-8)
    • Flutter version 2.3.0-1.0.pre.168 at /home/jpnurmi/Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 93242d7548 (7 days ago), 2021-04-28 19:19:05 -0700
    • Engine revision e7939e091e
    • Dart version 2.14.0 (build 2.14.0-48.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /home/jpnurmi/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_SDK_ROOT = /home/jpnurmi/Android/sdk
    • Java binary at: /home/jpnurmi/Android/studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Debian clang version 11.0.1-2
    • cmake version 3.18.4
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 4.1)
    • Android Studio at /home/jpnurmi/Android/studio
    • Flutter plugin version 55.0.1
    • Dart plugin version 201.9335
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3)
    • IntelliJ at /home/jpnurmi/JetBrains/IntelliJIdea
    • Flutter plugin version 53.1.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.55.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.22.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux
    • Chrome (web)    • chrome • web-javascript • Google Chrome 90.0.4430.93

• No issues found!
```
</details>

**Comment by jpnurmi** [comment]
A possible workaround:
```
ignore:
  - synthetic_package
```

**Comment by Salakar** [comment]
@jpnurmi could I see your `melos.yaml` configuration file if you have one? Thanks

**Comment by jpnurmi** [comment]
Here's the minimal one created by the _Detailed steps to reproduce the problem with gen_l10n_example from flutter/website_ in the issue description:
```
name: melos_gen_l10n
packages:
  - packages/**
environment:
  sdk: ">=2.12.0 <3.0.0"
```

**Comment by Salakar** [comment]
I see, `packages/**` would recursively bring in the flutter_gen ones too, I think we could automatically filter out packages contained in `.dart_tool`

**Comment by mrverdant13** [comment]
This is solved with #104, but it is not included in the `1.0.0-dev.1` prev release, which might be used to avoid stalling when using `build_runner`.
